src/registry/domain/s3.js

```diff
@@ -177,17 +171,17 @@ module.exports = function(conf) {
     if (fileInfo.gzip) {
       obj.ContentEncoding = 'gzip';
     }
-
-    getClient().putObject(obj, callback);
+    const upload = getClient().upload(obj);
+    upload.send(callback);
   };
 
   const putFile = (filePath, fileName, isPrivate, callback) => {
-    fs.readFile(filePath, (err, fileContent) => {
-      if (err) {
-        return callback(err);
-      }
-      putFileContent(fileContent, fileName, isPrivate, callback);
-    });
+    try {
+      const stream = fs.createReadStream(filePath);
+      putFileContent(stream, fileName, isPrivate, callback);
```

Comment:
Here we just create a stream reader instead of reading the whole content. For big files, this is actually necessary for not blocking the event loop with large GCs operations.